sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveParquetSourceSuite.scala

```diff
@@ -378,4 +378,15 @@ class HiveParquetSourceSuite extends ParquetPartitioningTest with ParquetTest {
       }
     }
   }
+
+  test("Create view with dashes in column type") {
+    sql(
+      """
+        |CREATE TABLE tbl (f array<struct<`date-of-creation`:string>>) USING parquet
+        |""".stripMargin)
+    sql(
+      """
+        |CREATE VIEW tblView AS SELECT f FROM tbl
```

Comment:
we can simply use the existing test case `CREATE VIEW t AS SELECT STRUCT('a' AS `$a`, 1 AS b) q`

Reply:
and let's make the test complete
```
withView("t") {
  sql("CREATE VIEW t AS SELECT STRUCT('a' AS $a, 1 AS b) q")
  checkAnswer(spark.table("t"), Row(Row("a", 1)))
}
```